### DISCUSSION TITLE
Shouldn't form onSubmit point to an arrow function?

### DISCUSSION THREAD

**Discussion by wondering639** [react-hook-form/react-hook-form#3918] [Q&A]
In the [documentation](https://react-hook-form.com/get-started)
`<form onSubmit={handleSubmit(onSubmit)}>`
is used, I would have expected an arrow function
`<form onSubmit={() => handleSubmit(onSubmit)}>`
to prevent execution during render.

Why isn't the arrow function used in the samples?

**Comment by bluebill1049** [comment]
this is what happening: `handleSubmit(onSubmit)(event)`

**Reply by wondering639** [reply]
Sorry, I don't understand yet ): (quite new to React/JavaScript). Can you give more details? Thanks :-)

**Reply by bluebill1049** [reply]
```
cons test = (e) => { console.log(e) }

<button type="button" onClick={test}>button</button>
```

try this code above.

**Reply by bluebill1049** [reply]
we want the `event` object in `handleSubmit`
